zerver/tests/test_push_notifications.py

```diff
@@ -791,6 +791,21 @@ def test_send_notification_endpoint_on_self_managed_plan(self) -> None:
         self.assert_json_error(result, "Your plan doesn't allow sending push notifications.")
         self.assertEqual(orjson.loads(result.content)["code"], "BAD_REQUEST")
 
+        # Check that sponsored realms are allowed to send push notifications.
+        remote_server.plan_type = RemoteRealm.PLAN_TYPE_COMMUNITY
+        remote_server.save()
+        result = self.uuid_post(
+            self.server_uuid,
+            "/api/v1/remotes/push/notify",
+            payload,
+            content_type="application/json",
+        )
+        self.assert_json_success(result)
+
+        # Reset the plan_type to test remaining cases.
+        remote_server.plan_type = RemoteRealm.PLAN_TYPE_SELF_MANAGED
+        remote_server.save()
```

Comment:
I am not sure whether this is the best place to add this check because the test name includes `self_managed_plan`.

Reply:
I think the placement is fine; I just changed the test name before merging.